guide/TEMPLATE.md

```diff
@@ -0,0 +1,53 @@
+# Using template from external repository
+
+## Repository structure
+
+The external repository must have a directory named ```template``` at the root directory, in which there are directories
+containing templates. The directory for each template can be freely named with alphanumeric characters and hyphen.
+
+Example:
+
+```
+template
+├── csharp
+│   ├── Dockerfile
+│   └── template.yml
+├── node
+│   ├── Dockerfile
+│   └── template.yml
+├── node-armhf
+│   ├── Dockerfile
+│   └── template.yml
+├── python
+│   ├── Dockerfile
+│   └── template.yml
+├── python-armhf
+│   ├── Dockerfile
+│   └── template.yml
+├── php-56
+│   ├── Dockerfile
+│   └── template.yml
+├── php-71
+│   ├── Dockerfile
+│   └── template.yml
+├── php-latest
+│   ├── Dockerfile
+│   └── template.yml
+└── ruby
+    ├── Dockerfile
+    └── template.yml
+```
+
+## Download external repository
+
+In order to build functions using 3rd party templates, you need to add 3rd templates before the build step, with the following command:
+
+```
+./faas-cli add-template https://github.com/alexellis/faas-cli
+```
```

Comment:
To avoid potential confusion for new users, should this be a genuine third party template repo, even if its just a copy of our current templates in an unfamiliar repo?

Reply:
That might be a good idea, but not a copy. Also s/alexellis/openfaas/g